commerce-cloud-code/app_yourshophere/cartridge/partials/pdp/prices.js

```diff
@@ -1,14 +1,50 @@
+/**
+ * @description PDP prices partial
+ * @param {Object} options - The options object
+ * @param {Object} options.product - The product object
+ * @returns {Object} The model object
+ */
 exports.createModel = function createModel(options) {
     const StringUtils = require('dw/util/StringUtils');
-
-    // @todo implement proper price logic
+    const PromotionMgr = require('dw/campaign/PromotionMgr');
+    const PriceBookMgr = require('dw/catalog/PriceBookMgr');
     const product = options.product.master ? options.product.variants[0] : options.product;
+    const priceModel = product.priceModel;
+
+    // find all list prices and use the highest one
+    const listPrices = PriceBookMgr.getSitePriceBooks().toArray()
+        .map((priceBook) => (priceBook.getParentPriceBook() && priceBook.getParentPriceBook().ID))
+        .filter((id) => id !== null)
+        .map((id) => priceModel.getPriceBookPrice(id));
+    listPrices.sort((a, b) => a.value - b.value);
+    const listPrice = listPrices.pop();
+
+    // @todo optimise performance for getActiveCustomerPromotions calls
+    const promotions = PromotionMgr.getActiveCustomerPromotions().getProductPromotions(product);
```

Comment:
I wouldn't loop - I would just calculate the first promo. If more then one promo are active for a product they will stack anyways